resharper/resharper-unity/src/Yaml/Psi/Resolve/MonoScriptReference.cs

```diff
@@ -69,7 +69,8 @@ public override ISymbolTable GetReferenceSymbolTable(bool useReferenceName)
             {
                 if (useReferenceName && resolvedName != null)
                 {
-                    if (resolvedName == candidate.ShortName)
+                    if (resolvedName == candidate.ShortName && !candidate.HasTypeParameters() 
+                        && candidate.GetContainingType() == null)
```

Comment:
The check for nested types is already handled in `UnityObjectPsiUtil.GetTypeElementCandidatesFromScriptAssetGuid`. It makes sense to put the type parameter check there, too, instead of here.